### PR TITLE
Update API tests to throw if no results returned

### PR DESCRIPTION
**Pull request by nklincoln**
Update API tests to assist with error diagnosis, and align object with DB index requirement

Signed-off-by: nkl199@yahoo.co.uk <nkl199@yahoo.co.uk>

### CODE REVIEW

File: benchmarks/api/fabric/lib/batch-create-asset.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ class BatchCreateAssetWorkload extends WorkloadModuleBase {
             docType: this.chaincodeID,
             content: '',
             creator: 'client' + this.workerIndex,
-            byteSize: this.byteSize
+            bytesize: this.byteSize
```
Comment (davidkel) [inline benchmarks/api/fabric/lib/batch-create-asset.js:51]: actually I thought `byteSize` (with a capital S) was better

File: src/fabric/api/fixed-asset/node/lib/fixed-asset.js [hunk 2]
```diff
@@ -27,10 +27,10 @@ class Asset extends Contract {
 
     /**
      * Return a null response
-     * @param {Context} ctx - the transaction context
+     * @param {Context} _ctx - the transaction context
```
Comment (davidkel) [inline src/fabric/api/fixed-asset/node/lib/fixed-asset.js:30]: If it's going to be changed would prefer it to be changed to `context`

File: src/fabric/api/fixed-asset/node/lib/fixed-asset.js [hunk 3]
```diff
@@ -27,10 +27,10 @@ class Asset extends Contract {
...
      * @returns {null}  a null response
      */
-    async emptyContract(ctx) {
+    async emptyContract(_ctx) {
```
Comment (davidkel) [inline src/fabric/api/fixed-asset/node/lib/fixed-asset.js:33]: As this is javascript should we just remove the parameter altogether ?

File: src/fabric/api/fixed-asset/node/lib/fixed-asset.js [hunk 4]
```diff
@@ -385,6 +353,36 @@ class Asset extends Contract {
         }
         return response;
     }
+
+    /**
+     * Get all results present in the iterator
+     * @param {Object} iterator the iterator to retrieve results from
+     * @returns {String[]} all results
```
Comment (davidkel) [inline src/fabric/api/fixed-asset/node/lib/fixed-asset.js:360]: add `@private` and `@async` ?

File: benchmarks/api/fabric/lib/batch-create-asset.js (hunk 1 above)
Comment (nklincoln) [reply benchmarks/api/fabric/lib/batch-create-asset.js]: Rolled the change through - touches many files as the requirement is for the parameter to match the specified DB index schema (consequently all the contracts that use it too)

**Review by davidkel — APPROVED** [review]

**Pull request merged**